modules/__tests__/describePush.js

```diff
@@ -32,6 +32,17 @@ const describePush = (createHistory) => {
 
         execSteps(steps, history, done)
       })
+
+      it('should trigger only once when path is changed', (done) => {
+        const spy = expect.createSpy()
+        const unlisten = history.listen(spy)
+        history.push('/test')
+        setTimeout(() => {
+          expect(spy.calls.length).toBe(1)
+          unlisten()
```

Comment:
Please catch errors and ensure that you call `unlisten` even if the assertion fails.